service/matching/handler/engine.go

```diff
@@ -515,7 +515,8 @@ pollLoop:
 		task, err := e.getTask(pollerCtx, taskListID, nil, taskListKind)
 		if err != nil {
 			// TODO: Is empty poll the best reply for errPumpClosed?
-			if errors.Is(err, tasklist.ErrNoTasks) || errors.Is(err, errPumpClosed) {
+			var taskListNotOwnedByHostError *cadence_errors.TaskListNotOwnedByHostError
+			if errors.Is(err, tasklist.ErrNoTasks) || errors.Is(err, errPumpClosed) || errors.As(err, &taskListNotOwnedByHostError) {
 				e.logger.Debug("no decision tasks",
```

Comment:
In addition, I think ` errors.As(err, &taskListNotOwnedByHostError)` should cause another logging message. "no decision task" really corresponds to ErrNoTask & errPumpClosed, but NotOwned is more like "wrong address, and we don't know if there are tasks there".

Reply:
As below, I'd actually return the host-is-wrong error all the way to frontend if possible, I'd not return emptyPollForDecisinoResponse

Reply:
Reverted this change - we just return the error and retry on frontend